### PR TITLE
Rtd integration of notebooks

### LINKED ISSUE #409: Example notebooks need to be become visible on our RTD page
* [x] Have you searched the issue tracker for the same problem?
* [x] Have you checked if you're using the latest version? If not, which version are you using?
* [x] Have you mentioned the steps to reproduce the issue?
* [x] Have you, if applicable, included error messages?

After #334 is done, `index.rst` needs to be modified such that examples are rendered on readthedocs (RTD) and therefore become visible. 

Typically, sphinx rendering only sees things below its root dir which is `./doc` in our case. Maybe one solution could be include a symbolic link there

i.e.

```bash
cd doc
ln -s ../examples .

# and then git it
```

### PR DESCRIPTION
**Pull request by fsenf**
I explore read-the-docs integration of our examples. I chose sphinx-gallery, see https://docs.readthedocs.io/en/stable/guides/jupyter.html#creating-galleries-of-examples-using-notebooks

This dev is branched off https://github.com/fsenf/tobac/tree/backintegration_of_tutorials and thus need to wait for #334 being back-integrated.

This fixes  #409 

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [ ] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Comment by w-k-jones** [comment]
This looks really nice! I'll review after #334 is merged

**Comment by fsenf** [comment]
Here is this direct link to the rendered gallery for your convenience: https://tobac--411.org.readthedocs.build/en/411/examples.html 

... there is room for improvement ... let's hope that some skilled people come around in future ;-)

**Comment by freemansw1** [comment]
@fsenf this is beautiful and a preeminent example of what our documentation *should* be. Let's get this in ASAP after #334 is in.

**Comment by JuliaKukulies** [comment]
Wow, very nice @fsenf ! Are you going to do the second review @freemansw1 ? If not, I can also review after #334

**Comment by fsenf** [comment]
OK, #334 is done! 

I updated this PR accordingly to hold also the newest changes of https://github.com/tobac-project/tobac/tree/RC_v1.5.x

@w-k-jones & @JuliaKukulies : It is ready for review now. Thank you for your willingness to review the PR!

**Review by w-k-jones — APPROVED** [review]
This looks really good! My only suggestions are regarding the naming of different sections, but I'll leave it open to your judgement whether you like these changes or not

**Comment by fsenf** [comment]
HI @JuliaKukulies , what is your opinion on this contribution?

File: doc/examples.rst
Comment (JuliaKukulies) [inline doc/examples.rst:21]:
```suggestion
   :caption: Examples of Using *tobac* with Observations
```

File: doc/examples.rst
Comment (JuliaKukulies) [inline doc/examples.rst:28]:
```suggestion
   :caption: Examples of Using *tobac* with Model Data
```

**Review by JuliaKukulies — APPROVED** [review]
This looks great - excellent contribution @fsenf!  Just a tiny syntax correction after your headlines update. Otherwise, the PR is ready to be merged.

**Comment by fsenf** [comment]
great! Thank you for your superfast feedback! I will merge it after testing is finished.

**Pull request merged**
